File: src/main/java/com/restaurant/entity/CommandeItem.java
package com.restaurant.entity;

public class CommandeItem {
    private int id;
    private Commande commande;
    private Burger burger;
    private Menu menu;
    private Complement complement;
    private int quantite;
    private double prixUnitaire;
    
    public CommandeItem() {}
    
    public CommandeItem(Commande commande, Burger burger, int quantite, double prixUnitaire) {
        this.commande = commande;
        this.burger = burger;
        this.quantite = quantite;
        this.prixUnitaire = prixUnitaire;
    }
    
    public CommandeItem(Commande commande, Menu menu, int quantite, double prixUnitaire) {
        this.commande = commande;
        this.menu = menu;
        this.quantite = quantite;
        this.prixUnitaire = prixUnitaire;
    }
    
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public Commande getCommande() {
        return commande;
    }
    
    public void setCommande(Commande commande) {
        this.commande = commande;
    }
    
    public Burger getBurger() {
        return burger;
    }
    
    public void setBurger(Burger burger) {
        this.burger = burger;
    }
    
    public Menu getMenu() {
        return menu;
    }
    
    public void setMenu(Menu menu) {
        this.menu = menu;
    }
    
    public Complement getComplement() {
        return complement;
    }
    
    public void setComplement(Complement complement) {
        this.complement = complement;
    }
    
    public int getQuantite() {
        return quantite;
    }
    
    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }
    
    public double getPrixUnitaire() {
        return prixUnitaire;
    }
    
    public void setPrixUnitaire(double prixUnitaire) {
        this.prixUnitaire = prixUnitaire;
    }
    
    public double getSousTotal() {
        return quantite * prixUnitaire;
    }
    
    public String getProduitNom() {
        if (burger != null) return burger.getNom();
        if (menu != null) return menu.getNom();
        if (complement != null) return complement.getNom();
        return "Inconnu";
    }
    
    @Override
    public String toString() {
        return "CommandeItem{" +
               "produit=" + getProduitNom() +
               ", quantite=" + quantite +
               ", prixUnitaire=" + prixUnitaire +
               ", sousTotal=" + getSousTotal() +
               '}';
    }
}

File: src/Entity/CommandeItem.php
<?php

namespace App\Entity;

use App\Repository\CommandeItemRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CommandeItemRepository::class)]
#[ORM\Table(name: 'commande_items')]
class CommandeItem
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Commande::class, inversedBy: 'commandeItems')]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private ?Commande $commande = null;

    // #[ORM\ManyToOne(targetEntity: Burger::class)]
    #[ORM\ManyToOne(targetEntity: Burger::class, inversedBy: 'commandeItems')]
    private ?Burger $burger = null;

    #[ORM\ManyToOne(targetEntity: Menu::class)]
    private ?Menu $menu = null;

    #[ORM\ManyToOne(targetEntity: Complement::class)]
    private ?Complement $complement = null;

    #[ORM\Column]
    private ?int $quantite = 1;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prixUnitaire = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    // Getters and setters pour toutes les propriétés...
    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): self
    {
        $this->commande = $commande;
        return $this;
    }

    public function getBurger(): ?Burger
    {
        return $this->burger;
    }

    public function setBurger(?Burger $burger): self
    {
        $this->burger = $burger;
        return $this;
    }

    public function getMenu(): ?Menu
    {
        return $this->menu;
    }

    public function setMenu(?Menu $menu): self
    {
        $this->menu = $menu;
        return $this;
    }

    public function getComplement(): ?Complement
    {
        return $this->complement;
    }

    public function setComplement(?Complement $complement): self
    {
        $this->complement = $complement;
        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;
        return $this;
    }

    public function getPrixUnitaire(): ?string
    {
        return $this->prixUnitaire;
    }

    public function setPrixUnitaire(string $prixUnitaire): self
    {
        $this->prixUnitaire = $prixUnitaire;
        return $this;
    }

    public function getTotal(): ?string
    {
        if ($this->prixUnitaire === null || $this->quantite === null) {
            return null;
        }
        return (string)($this->prixUnitaire * $this->quantite);
    }
}